protocol/app/flags/flags.go

```diff
@@ -49,6 +59,15 @@ func AddFlagsToCmd(cmd *cobra.Command) {
 	)
 }
 
+// Validate checks that the flags are valid.
+func (f *Flags) Validate() error {
+	// Validtors must have cosmos grpc services enabled.
+	if !f.NonValidatingFullNode && !f.GrpcEnable {
+		return fmt.Errorf("grpc.enable must be set to true - validating requires gRPC server")
```

Comment:
Nit: explicitly say sth like:
`grpc.enable must be set to true when 'non-validating-full-node' flag is provided. `